src/nvim/ops.c

```diff
@@ -2874,7 +2872,8 @@ void do_put(int regname, yankreg_T *reg, int dir, long count, int flags)
 
     /* adjust '] mark */
     curbuf->b_op_end.lnum = curwin->w_cursor.lnum - 1;
-    curbuf->b_op_end.col = bd.textcol + totlen - 1;
+    assert(totlen <= INT_MAX);
```

Comment:
just change the cast to `(colnr_T)(bd.textcol + totlen - 1)` and it would be fine?